open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -49,7 +72,432 @@ def get_available_docker_host(self, req_count, hackathon):
+                                       host_base=self.util.safe_get_config('dockerhostserver.storage.host_base',
+                                                                           '.blob.core.chinacloudapi.cn'))
+            try:
+                blob_service.get_container_metadata(container_name)
+                return True, storage.service_name, container_name
+            except Exception as e:
+                if e.message != 'Not found (The specified container does not exist.)':
+                    self.log.error('Encounter an error when checking container:%s ' % container_name)
+                    self.log.error(e)
+                    continue
+            try:
+                blob_service.create_container(container_name=container_name, x_ms_blob_public_access='container')
+                return True, storage.service_name, container_name
+            except Exception as e:
+                self.log.error('Encounter an error when creating container:%s ' % container_name)
+                self.log.error(e)
+        return False, None, None
+
+    def __get_available_cloud_service(self, hackathon_id):
+        """
+        Get available cloud service
+
+        :param hackathon_id: the id of hackathon
+        :type hackathon_id: integer
+
+        :return: if there is a available cloud service, then return (True, cloud service name). Otherwise, return
+                (False, None).
+        :rtype: 2-element tuple: (bool, str|unicode)
+        """
+        config_name = self.util.safe_get_config('dockerhostserver.azure.cloud_service.name',
+                                                'dockerhostprivatecloudservice')
+        service_name = config_name + str(hackathon_id)
+        service_location = self.util.safe_get_config('dockerhostserver.azure.cloud_service.location', 'China East')
+        sms = self.__get_sms_object(hackathon_id)
+        try:
+            sms.get_hosted_service_properties(service_name, True)
+            return True, service_name
+        except Exception as e:
+            if e.message != 'Not found (Not Found)':
+                self.log.error('Encounter an error when checking cloud service:%s ' % service_name)
+                self.log.error(e)
+                return False, None
+        # not found, then create one
+        try:
+            if sms.check_hosted_service_name_availability(service_name=service_name).result:
+                sms.create_hosted_service(service_name=service_name, label=service_name, location=service_location)
+                return True, service_name
+            else:
+                self.log.error('Cloud service name:%s is not available. ' % service_name)
+                return False, None
+        except Exception as e:
+            self.log.error('Encounter an error when creating cloud service:%s ' % service_name)
+            self.log.error(e)
+        return False, None
+
+    def __get_customdata_from_local_file(self):
```

Comment:
Where is the shell file? Suppose you should include it in some sub directory of our git repo. And the default path a relative path to the root of repository